Source/missiles.cpp

```diff
@@ -1478,48 +1478,25 @@ void __fastcall SetMissDir(int mi, int dir)
 // check for example the calls in `InitMonsterGFX`
 void __fastcall LoadMissileGFX(BYTE mi)
 {
-	MisFileData *v1;   // esi
-	unsigned char *v2; // eax
-	signed int v3;     // ecx
-	int *v4;           // edx
-	int v5;            // edi
-	unsigned char v6;  // cl
-	int v7;            // eax
-	_DWORD *v8;        // edi
-	int v9;            // ebx
-	char arglist[256]; // [esp+8h] [ebp-100h]
-
-	v1 = &misfiledata[(unsigned char)mi];
-	if (v1->mFlags & MFLAG_ALLOW_SPECIAL) {
-		sprintf(arglist, "Missiles\\%s.CL2", v1->mName);
-		v2 = LoadFileInMem(arglist, 0);
-		v3 = 0;
-		if (v1->mAnimFAmt) {
-			v4 = (int *)v1->mAnimData;
-			do {
-				v5 = (int)&v2[*(_DWORD *)&v2[4 * v3++]];
-				*v4 = v5;
-				++v4;
-			} while (v3 < v1->mAnimFAmt);
-		}
+	char pszName[256];
+	int i;
+	BYTE *file;
+	MisFileData *mfd = &misfiledata[mi];
+	if (mfd->mFlags & MFLAG_ALLOW_SPECIAL) {
+		sprintf(pszName, "Missiles\\%s.CEL", mfd->mName);
+		file = LoadFileInMem(pszName, 0);
+		for (i = 0; i < mfd->mAnimFAmt; i++)
+			mfd->mAnimData[i] = &file[*(_DWORD *)&file[4 * i]];

```

Comment:
Does this work?
```suggestion
			mfd->mAnimData[i] = &file[file[4 * i]];
```

Reply:
`&file[file[4 * i]];` doesn't work

Reply:
`mfd->mAnimData[i] = &file[*(DWORD *)&file[4 * i]];` works, can you suggest such a change?

Reply:
Applied the trick from InitMonsterGFX instead